python/plugins/processing/algs/qgis/Transect.py

```diff
@@ -0,0 +1,187 @@
+        self.distance = None
+        self.angle = None
+        self.side = None
+
+    def initParameters(self, config=None):
+        self.sides = [self.tr('Left'),
+                      self.tr('Right'),
+                      self.tr('Both')]
+        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
+                                                              self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterNumber(self.DISTANCE,
+                                                       self.tr('Distance'),
+                                                       type=QgsProcessingParameterNumber.Double,
+                                                       defaultValue=5.0))
+        self.addParameter(QgsProcessingParameterNumber(self.ANGLE,
+                                                       self.tr('Angle'),
+                                                       type=QgsProcessingParameterNumber.Double,
+                                                       defaultValue=90.0))
+        self.addParameter(QgsProcessingParameterEnum(self.SIDE,
+                                                     self.tr('Side'), options=self.sides))
+
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Transect')))
+
+    def name(self):
+        return 'transect'
+
+    def displayName(self):
+        return self.tr('Transect')
+
+    def outputName(self):
+        return self.tr('Transect')
+
+    def inputLayerTypes(self):
+        return [QgsProcessing.TypeVectorLine]
+
+    def processAlgorithm(self, parameters, context, feedback):
+        source = self.parameterAsSource(parameters, self.INPUT, context)
+        distance = self.parameterAsDouble(parameters, self.DISTANCE, context)
+        angle = self.parameterAsDouble(parameters, self.ANGLE, context)
+        side = self.parameterAsEnum(parameters, self.SIDE, context)
+        fields = source.fields()
+        fields.append(QgsField("TR_FID", QVariant.Int))
+        fields.append(QgsField("TR_ID", QVariant.Int))
+        fields.append(QgsField("TR_SEGMENT", QVariant.Int))
+        fields.append(QgsField("TR_ANGLE", QVariant.Double))
+        fields.append(QgsField("TR_LENGTH", QVariant.Double))
+        fields.append(QgsField("TR_ORIENT", QVariant.String))
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT, context,
+                                               fields, QgsWkbTypes.LineString, source.sourceCrs())
+
+        features = source.getFeatures()
+        total = 100.0 / source.featureCount() if source.featureCount() else 0
+
+        number = 0
+        for current, f in enumerate(features):
+            if feedback.isCanceled():
+                break
+
+            if not f.hasGeometry():
+                pass
+            else:
+                input_geometry = f.geometry()
+                if input_geometry:
```

Comment:
This check is redundant - the `f.hasGeometry()` check above has just done the same thing